Base lower bounds are wrong

Compiling HLS with GHC-9.0.2 results in compilation errors:

```
[ 6 of 76] Compiling Development.IDE.GHC.Compat.Core ( src/Development/IDE/GHC/Compat/Core.hs, dist/build/Development/IDE/GHC/Compat/Core.o, dist/build/Development/IDE/GHC/Compat/Core.dyn_o )

src/Development/IDE/GHC/Compat/Core.hs:509:1: error:
    Could not load module `GHC.Driver.Env'
    It is a member of the hidden package `ghc-lib-parser-9.4.8.20231111'.
    Perhaps you need to add `ghc-lib-parser' to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
```

Unbuildable configurations must be disallowed by the cabal file, so we don't revive cabal hell.

https://github.com/haskell/haskell-language-server/blob/0a02832ed3d2ef8b5128b1f8de2cbdc98eb4bc2f/haskell-language-server.cabal#L17

https://github.com/haskell/haskell-language-server/blob/0a02832ed3d2ef8b5128b1f8de2cbdc98eb4bc2f/haskell-language-server.cabal#L38

--- a/ghcide/ghcide.cabal
+++ b/ghcide/ghcide.cabal
@@ -73,7 +73,7 @@ library
     , filepath
     , fingertree
     , focus                        >=1.0.3.2
-    , ghc                          >=9.0
+    , ghc                          >=9.2
     , ghc-boot
     , ghc-boot-th
     , ghc-check                    >=0.5.0.8

--- a/ghcide/ghcide.cabal
+++ b/ghcide/ghcide.cabal
@@ -14,7 +14,7 @@ homepage:
   https://github.com/haskell/haskell-language-server/tree/master/ghcide#readme
 
 bug-reports:        https://github.com/haskell/haskell-language-server/issues
-tested-with:        GHC ==9.0.2 || ==9.2.5
+tested-with:        GHC == 9.8.1 || ==9.6.4 || ==9.4.8 || ==9.2.8
 extra-source-files:
   CHANGELOG.md
   README.md
@@ -52,7 +52,7 @@ library
     , aeson
     , array
     , async
-    , base                         >=4        && <5
+    , base                         >=4.16     && <5
     , base16-bytestring            >=0.1.1    && <1.1
     , binary
     , bytestring
@@ -275,7 +275,7 @@ executable ghcide
   -- Enable collection of heap statistics
   main-is:            Main.hs
   build-depends:
-    , base                  >=4 && <5
+    , base                  >=4.16 && <5
     , data-default
     , extra
     , ghcide

--- a/haskell-language-server.cabal
+++ b/haskell-language-server.cabal
@@ -14,7 +14,7 @@ copyright:          The Haskell IDE Team
 license:            Apache-2.0
 license-file:       LICENSE
 build-type:         Simple
-tested-with:        GHC == 9.0.2 || ==9.2.5
+tested-with:        GHC == 9.8.1 || ==9.6.4 || ==9.4.8 || ==9.2.8
 extra-source-files:
   README.md
   ChangeLog.md
@@ -35,7 +35,7 @@ source-repository head
 
 common common-deps
   build-depends:
-    , base       >=4.12 && <5
+    , base       >=4.16 && <5
     , directory
     , extra
     , filepath
@@ -593,6 +593,7 @@ test-suite wrapper-test
   main-is:            Main.hs
 
 benchmark benchmark
+    import:           common-deps
     -- Depends on shake-bench which is unbuildable after this point
     if impl(ghc >= 9.5)
       buildable: False
@@ -622,12 +623,8 @@ benchmark benchmark
 
     build-depends:
         aeson,
-        base == 4.*,
         containers,
         data-default,
-        directory,
-        extra,
-        filepath,
         ghcide-bench,
         haskell-language-server,
         hls-plugin-api,
@@ -636,5 +633,4 @@ benchmark benchmark
         optparse-applicative,
         shake,
         shake-bench == 0.2.*,
-        text,
         yaml

